translate/google/cloud/translate_v2/_http.py

```diff
@@ -29,15 +29,17 @@ class Connection(_http.JSONConnection):
     :param client_info: (Optional) instance used to generate user agent.
     """
 
-    def __init__(self, client, client_info=None):
+    DEFAULT_API_ENDPOINT = "https://translation.googleapis.com"
+
+    def __init__(self, client, client_info=None, api_endpoint=None):
+        if api_endpoint is None:
+            api_endpoint = self.DEFAULT_API_ENDPOINT
+        Connection.API_BASE_URL = api_endpoint
```

Comment:
Better to overwrite on the instance *after* calling `super()` (to avoid polluting the class), e.g.:

```suggestion
        super(Connection, self).__init__(client, client_info)
        self.API_BASE_URL = api_endpoint
```

Reply:
I did this because there's a class method in `JSONConnection` that uses this URL. Is there another way to override it?

https://github.com/googleapis/google-cloud-python/blob/1dc92c79d53c566968bd6767ab1d7d3316db9ab9/core/google/cloud/_http.py#L180-L212

Reply:
Hmm, that `@classmethod` looks a little bogus to me.  All actual uses of it are via instances:

```bash
$ git grep "\.build_api_url"
bigquery/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
bigquery/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", {"bar": "baz"})
bigquery/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
bigquery/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
core/google/cloud/_http.py:        url = self.build_api_url(
core/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
core/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", {"bar": "baz", "qux": ["quux", "corge"]})
dns/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
dns/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", {"bar": "baz"})
dns/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
logging/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
resource_manager/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
resource_manager/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", {"bar": "baz"})
resource_manager/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
runtimeconfig/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
storage/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
storage/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
storage/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", {"bar": "baz"})
translate/tests/unit/test__http.py:        self.assertEqual(conn.build_api_url("/foo"), URI)
translate/tests/unit/test__http.py:        uri = conn.build_api_url("/foo", query_params=query_params)
translate/tests/unit/test__http.py:        expected_uri = conn.build_api_url("/rainbow")
```

Reply:
Would it be alright to switch this to an instance method in core's `_http.py` then?

Reply:
I would prefer that.